ros2topic/ros2topic/verb/pub.py

```diff
@@ -155,11 +155,16 @@ def timer_callback():
             print('publishing #%d: %r\n' % (count, msg))
         pub.publish(msg)
 
-    timer = node.create_timer(period, timer_callback)
-    while times == 0 or count < times:
-        rclpy.spin_once(node)
-
-    # give some time for the messages to reach the wire before exiting
-    time.sleep(keep_alive)
-
-    node.destroy_timer(timer)
+    # give some time for discovery process
+    time.sleep(0.1)
+    timer_callback()
+    if times != 1:
+        timer = node.create_timer(period, timer_callback)
```

Comment:
Or we could have this option in rclcpp and rclpy `Timer` class? probably not, since user can do the same in application code just like this change.

Reply:
If I understand correctly, if we want to make the timer fire immediately, we should make that change in `rcl`. That being said, maybe this change helps `ros2 topic pub` more in line with a user's expectations.